### PR TITLE
MAGN-8974 "Show Grid" no longer works after orbiting the grid upside down

### PR DESCRIPTION
**Pull request by marimano**
### Purpose

When the grid of 3D background is in front of any other geometry, toggling grid visibility does not cause to redraw the background.

Also, grid visibility flag is set to false when the background become inactive similarly to turning off "Navigate Background 3D Preview":
inactive_background

### Reviewers

@ikeough

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 1]
```diff
@@ -1507,14 +1507,14 @@ internal bool CanToggleFullscreenWatchShowing(object parameter)
...
         {
-            if (!CanToggleBackgroundGridVisibility(parameter)) return;
+            if (BackgroundPreviewViewModel == null || !BackgroundPreviewViewModel.Active) return;
 
             BackgroundPreviewViewModel.IsGridVisible = !BackgroundPreviewViewModel.IsGridVisible;
         }
 
         internal bool CanToggleBackgroundGridVisibility(object parameter)
         {
-            return BackgroundPreviewViewModel != null && BackgroundPreviewViewModel.Active;
```
Comment (marimano) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: `IsEnabled` of corresponding menu item is already bound to `BackgroundPreviewViewModel.Active`. So this is redundant

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 2]
```diff
@@ -944,24 +951,27 @@ private void DrawGrid()
...
+                    Model3DDictionary[DefaultGridName] = gridModel3D;
+                }
+                else
+                {
+                    Model3DDictionary.Remove(DefaultGridName);
+                }
             }
 
-            RaisePropertyChanged("SceneItems");
+            OnSceneItemsChanged();
```
Comment (marimano) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: Re-evaluate SceneItems with redrawing

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 3]
```diff
@@ -704,6 +705,12 @@ internal void ComputeFrameUpdate()
         #endregion
 
         #region private methods
+
+        private void OnSceneItemsChanged()
```
Comment (ikeough) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: Nice.

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 4]
```diff
@@ -135,10 +135,14 @@ protected override void OnActiveStateChanged()
...
 
-            if (active == false && CanNavigateBackground)
+            if (active) return;
+
+            if (CanNavigateBackground)
             {
                 CanNavigateBackground = false;
             }
+
+            IsGridVisible = false;
```
Comment (ikeough) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: Why are we setting `IsGridVisible` to false when the `Active` state changes?

**Comment by ikeough** [comment]
@marimano This PR says 8974 at the top, but seems to relate to 8797. Does this PR solve both issues? If so, how? I have only one small question as it relates to 8797, then LGTM.

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 5]
```diff
@@ -944,24 +951,27 @@ private void DrawGrid()
...
-                Model3DDictionary[DefaultGridName] = gridModel3D;
-            }
-            else
-            {
-                Model3DDictionary.Remove(DefaultGridName);
+                    Model3DDictionary[DefaultGridName] = gridModel3D;
+                }
+                else
+                {
+                    Model3DDictionary.Remove(DefaultGridName);
```
Comment (ikeough) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: We do not need to remove the grid from the scene - unless we have to. You can use `gridModel3D.Visibility == Visibility.Hidden`. That will take care of making the object invisible in the scene. Do we have to remove it from the scene?

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs (hunk 4 above)
Comment (marimano) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: `IsGridVisible = false` when `Active` is false only

**Comment by marimano** [comment]
@ikeough, 8974 is solved by refreshing view after toggling grid, 8797 is solved by setting `IsGridVisible` to false when background becomes inactive (`active=false`)

**Comment by marimano** [comment]
Also, I have replaced adding/removing grid with setting visibility value of it

**Comment by ikeough** [comment]
@marimano Thanks. LGTM. 

And merging...

**Pull request merged**